core/indexer/elasticsearch.go

```diff
@@ -358,28 +359,50 @@ func (ei *elasticIndexer) buildTransactionBulks(
 			}
 
 			currentTx, ok := currentTxHandler.(*transaction.Transaction)
-			if !ok {
-				ei.logger.Warn("elasticsearch found tx in pool but of wrong type")
+			if ok && currentTx != nil {
+				bulks[currentBulk] = append(bulks[currentBulk], &Transaction{
+					Hash:          hex.EncodeToString(txHash),
+					MBHash:        hex.EncodeToString(mbHash),
+					BlockHash:     hex.EncodeToString(blockHash),
+					Nonce:         currentTx.Nonce,
+					Value:         currentTx.Value,
+					Receiver:      hex.EncodeToString(currentTx.RcvAddr),
+					Sender:        hex.EncodeToString(currentTx.SndAddr),
+					ReceiverShard: mb.ReceiverShardID,
+					SenderShard:   mb.SenderShardID,
+					GasPrice:      currentTx.GasPrice,
+					GasLimit:      currentTx.GasLimit,
+					Data:          currentTx.Data,
+					Signature:     hex.EncodeToString(currentTx.Signature),
+					Timestamp:     time.Duration(header.GetTimeStamp()),
+					Status:        mbTxStatus,
+				})
+				continue
+			}
+
+			currentSc, ok := currentTxHandler.(*smartContractResult.SmartContractResult)
```

Comment:
Do these txs include the transactions for the rewards?

Reply:
No they are not implemented yet. Maybe a more uniform solution can be implemented when we have all tx types